### PR TITLE
Document debug configuration setting

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

A new `debug` setting is available to enable Home Assistant's built-in debug, which can help locate misbehaving integrations by enabling run-time checks for implementation errors. It can block many unsafe thread operations from crashing the system. Enabling debug has a slight performance impact on the system and is not recommended for long-term use.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/115390
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/configuration/basic.markdown [hunk 1]
```diff
@@ -110,6 +111,10 @@ country:
   description: "Country in which Home Assistant is running. This may, for example, influence radio settings to comply with local regulations. The country should be specified as an ISO 3166.1 alpha-2 code. Pick your country from the column **Code** of [Wikipedia's list of ISO 31661 alpha-2 officially assigned code codes](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#Officially_assigned_code_elements)"
   required: false
   type: string
+debug:
+  description: Enable Home Assistant's built-in debug, which can help locate misbehaving integrations by enabling run-time checks for implementation errors. It can block many unsafe thread operations from crashing the system. Enabling debug has a slight performance impact on the system and is not recommended for long-term use.
+  required: false
+  type: boolean
```
Comment (bdraco) [inline source/_docs/configuration/basic.markdown]:
```suggestion
  type: boolean
  default: false
```

**Comment by bdraco** [comment]
Probably should update the troubleshooting page as well with a what to do if Home Assistant is crashing.

Debug is a good start for that, and we can expand it in the future with the profiler integration to help find memory leaks

**Comment by bdraco** [comment]
need to restore the missing changes from the conflict with https://github.com/home-assistant/home-assistant.io/pull/32371

**Review by MartinHjelmare — APPROVED** [review]

**Comment by bdraco** [comment]
thanks

**Pull request merged**